src/Surface.zig

```diff
@@ -4244,6 +4244,39 @@ pub fn performBindingAction(self: *Surface, action: input.Binding.Action) !bool
             screen.dirty.selection = true;
             try self.queueRender();
         },
+
+        .copy_url_to_clipboard => {
+            if (!self.mouse.over_link) {
+                return false;
+            }
+
+            const pos = try self.rt_surface.getCursorPos();
+            const point = self.posToViewport(pos.x, pos.y);
+            const screen = &self.renderer_state.terminal.screen;
+
+            const pin = screen.pages.pin(.{ .viewport = point }) orelse {
+                return false;
+            };
```

Comment:
@liby We should probably leave a log if failed to get a pin for the clicked point.

log.warn("failed to get pin for clicked point", .{});